### ISSUE TITLE
[Feature Request] focused_window displays the current mark

### ISSUE THREAD

**Issue by travankor** [greshake/i3status-rust#486]
The `focused_window` block does not change when I send a `title_format` command to a window on sway (tested via `swaymsg` or the config file). The GUI header that sway keeps on top of the window does however change as expected.

`i3status-rs` sticks with `%title`.

**Comment by ammgws** [comment]
Does it happen on `i3` too?

Looks like the `focused_window` block is using `name` to get the window title, but I checked the output of `swaymsg -t get_tree` and it doesn't look like this field is updated by sway (and i3?) when using title_format.

**Comment by travankor** [comment]
I'm not sure since I have barely used i3. Although I think you're right, in which case, there is no bug in `i3status-rs`. (This sort of feels like a glaring design flaw in i3 tho.)

I guess I could work around this by using mark command. Perhaps `i3status-rs` can add a block that shows the mark of the focused window?

**Comment by atheriel** [comment]
I'm open to expanding the functionality of the `focused_window` block in this manner.

**Comment by ammgws** [comment]
Looks like marks aren't shown in the get_tree output so might not be trivial to implement.

Looks like can get marks using get_marks, though.

So add an option to the block to display the value of mark instead of title_format, and then when focus changes call get_tree and see if the container has a mark, and if so used that.

**Comment by ohk2kt3t4** [comment]
@travankor How are you using marks to get around this?

**Comment by travankor** [comment]
It's not really a replacement for updating title_format, but more of a separate request I guess.
Both the title and the mark are shown on the header bar, but I usually hide the header bar by default.

Feel free to close if it's out of scope for the project.

**Comment by ammgws** [comment]
It might be worth considering if it's not too hard to implement. Could you please show a screenshot of the header bar when they are both showing? I don't use marks so not sure it looks like.

**Comment by travankor** [comment]
>It might be worth considering if it's not too hard to implement. Could you please show a screenshot of the header bar when they are both showing? I don't use marks so not sure it looks like.

Something like [this](https://files.catbox.moe/d360ia.png) although this is just an example. Marks are unique for each window, but you can have several marks per window in theory.

**Comment by ammgws** [comment]
@travankor Implemented this in #532, however need to wait for the i3ipc crate to merge a PR first before we can add the feature. I have tested it and it seems to work OK, would you like to try #532 and see if it works for you?

**Comment by ammgws** [comment]
Implemented via #532
